core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -434,6 +439,7 @@ public Location add(Operation operation) throws IOException {
                 ensureOpen();
                 Location location = current.add(bytes);
                 assert assertBytesAtLocation(location, bytes);
+                lastWriteLocation = location;
```

Comment:
this assignment must be checked. We are under a readLock and that means we can have a stale write here. I think it's simpler to fetch this when needed lazily and don't have a member?